src/main/java/duke/duke/TaskList.java

```diff
@@ -0,0 +1,177 @@
+     *
+     * @param input The respective index of the ArrayList needed to be mark or unmarked.
+     * @param arr The ArrayList with the current list of tasks.
+     * @throws NumberFormatException If input is less than or equals to zero.
+     */
+
+    public void mark(String input, ArrayList<Task> arr) {
+        try {
+            String s = input.replaceAll("\\D+", "");
+            int clean = Integer.parseInt(s) - 1; // Parse to find what number in list to toggle
+            // Edge Case
+            if (clean > arr.size()) {
+                System.out.println("Error! No tasked added");
+            } else if (input.toCharArray()[0] != 'u') {
+                arr.get(clean).setMarked();
+                System.out.println("Nice! I've marked this task as done:\n "
+                        + "   " + arr.get(clean).toString());
+            } else {
+                arr.get(clean).setUnmarked();
+                System.out.println("OK, I've marked this task as not done yet:\n "
+                        + "   " + arr.get(clean).toString());
+            }
+        } catch (NumberFormatException e) {
+            System.out.println("Don't be cheeky. Please write something that makes sense.");
+        }
+    }
+
+    /**
+     * Removes the respective index of the task in the ArrayList given by the input.
+     * Prints an acknowledgement if successful, else prints an edge case warning.
+     *
+     * @param input The respective index of the ArrayList needed to be removed.
+     * @param arr The ArrayList with the current list of tasks.
+     * @throws NumberFormatException If input is less than or equals to zero.
+     */
+
+    public void delete(String input, ArrayList<Task> arr) {
+        try {
+            String s = input.replaceAll("\\D+", "");
+            int delete = Integer.parseInt(s) - 1; // Parse to find what number in list to delete
+            // Edge Case
+            if (delete > arr.size()) {
+                System.out.println("Error! Nothing to delete!");
+            } else {
+                System.out.println("Noted. I've removed this task: \n   "
+                       + arr.get(delete).toString());
+                arr.remove(delete);
+                System.out.println("Now you have " + arr.size() + " tasks in the list.");
+            }
+        } catch (Exception e) {
+            System.out.println("Error! Nothing to delete!");
+        }
+    }
+
+    /**
+     * Creates a new Deadline class with the input detailing the class variables.
+     * Adds the new Deadline class to the ArrayList.
+     * Prints an acknowledgement if successful.
+     *
+     * @param input The information required for the initialization of the Deadline class.
+     * @param arr The ArrayList with the current list of tasks.
+     * @throws IndexOutOfBoundsException If there are no details in input.
+     */
+
+    public void deadline(String input, ArrayList<Task> arr) {
```

Comment:
May want to consider renaming to a verb such as "setDeadline" instead of just "deadline"?